File: 3345 - Smallest Divisible Digit Product I/Method 1/solution.java
class Solution {
    public int smallestNumber(int n, int t) {
        int ans = 0;
        for(int i = n; i<=n+10; i++) {
            int prod = product(i);
            if(prod%t==0) {
                ans = i;
                break;
            }
        }
        return ans;
    }
    public int product(int m) {
        int prod = 1;
        while(m>0) {
            int rem = m % 10;
            prod = prod * rem;
            m = m / 10;
        }
        return prod;
    }
}

File: 3345 - Smallest Divisible Digit Product I/Method 1/solution.js
/**
 * @param {number} n
 * @param {number} t
 * @return {number}
 */
var smallestNumber = function(n, t) {
    let ans = 0;
    for(let i = n; i<=n+10; i++) {
        let prod = product(i);
        if(prod%t==0) {
            ans = i;
            break;
        }
    }
    return ans;
};
var product = function(m) {
    let prod = 1;
    while(m>0) {
        let rem = m % 10;
        prod = prod * rem;
        m = Math.floor(m/10);
    }
    return prod;
}